### PR TITLE
Add Eslint CI action

### PR DESCRIPTION
**Pull request by bfcarpio**
Adds a small ESLint check to a Pipeline. I'm sure somewhere in your settings you can say that PRs are blocked unless this check passes.

Also ignores the one error that was being picked up because I couldn't actually find a way to fix it.

Explicitly adds a dependency that was mentioned in the ESLint configuration and removes `--fix` for consistency. Changes in the pipeline don't get applied to the PR.

Happy to work with you on adding or removing changes as desired.

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-node@v3
+      with:
+        node-version: 18
+        cache: 'npm'
+    - run: npm i --include=dev
```
Comment (duyhai) [inline .github/workflows/node.js.yml]: Can we use yarn here too? I'd prefer to make this consistent with the dev flow
https://github.com/Borales/actions-yarn

File: package.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "dev": "bash scripts/dev_env.sh",
     "start": "ts-node src/index.ts",
-    "lint": "eslint --fix . --ext .ts",
+    "lint": "eslint . --ext .ts",
```
Comment (duyhai) [inline package.json:9]:
```suggestion
    "lint": "eslint . --ext .ts",
    "lint:fix": "eslint --fix . --ext .ts",

```

**Review by duyhai — COMMENTED** [review]
Awesome! Stoked to see the first PR from another 1.5 member :)

File: package.json (hunk 2 above)
Comment (bfcarpio) [reply package.json:9]: Fixed in a fancy way

File: .github/workflows/node.js.yml (hunk 1 above)
Comment (bfcarpio) [reply .github/workflows/node.js.yml]: I did my best O.O

File: .github/workflows/node.js.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+      with:
+        node-version: 18
+        cache: 'yarn'
+    - name: Install
+      uses: borales/actions-yarn@v4
+      with:
+        cmd: install
+    - name: Lint
+      uses: borales/actions-yarn@v4
+        cmd: run lint
```
Comment (duyhai) [inline .github/workflows/node.js.yml]:
```suggestion
        cmd: lint
```

File: package.json [hunk 4]
```diff
@@ -6,7 +6,8 @@
   "scripts": {
     "dev": "bash scripts/dev_env.sh",
     "start": "ts-node src/index.ts",
-    "lint": "eslint --fix . --ext .ts",
+    "lint": "eslint . --ext .ts",
+    "lint:fix": "npm run lint -- --fix",
```
Comment (duyhai) [inline package.json]:
```suggestion
    "lint:fix": "yarn lint -- --fix",

```

**Review by duyhai — APPROVED** [review]
Thanks! LGTM

**Pull request merged**